src/Libraries/Microsoft.Extensions.Http.Diagnostics/Logging/Internal/LoggerMessageStateExtensions.cs

```diff
@@ -26,8 +27,7 @@ public static void AddRequestHeaders(this LoggerMessageState state, List<KeyValu
         {
             var key = _requestPrefixedNamesCache.GetOrAdd(
                 items[i].Key,
-                static (x, p) => p + Normalize(x),
-                HttpClientLoggingTagNames.RequestHeaderPrefix);
+                static x => HttpClientLoggingTagNames.RequestHeaderPrefix + Normalize(x));
```

Comment:
Will this transformation lead to a delegate allocation on each invocation of GetOrAdd?

Reply:
No, the compiler is smart enough to create a `static` backing field and assign the delegate instance to it just once:
```cs
string key = LoggerMessageStateExtensions._requestPrefixedNamesCache.GetOrAdd(items[i].Key, LoggerMessageStateExtensions.<>c.<>9__2_0 ?? (LoggerMessageStateExtensions.<>c.<>9__2_0 = new Func<string, string>((object) LoggerMessageStateExtensions.<>c.<>9, __methodptr(<AddRequestHeaders>b__2_0))));

//...
    [CompilerGenerated]
    [Serializable]
    private sealed class <>c
    {
      [Nullable(0)]
      public static readonly LoggerMessageStateExtensions.<>c <>9;
      [Nullable(new byte[] {0, 1, 1})]
      public static Func<string, string> <>9__2_0;
      // ...
    }
```